File: ruby/lib/ccpayment/services/swap_service.rb
module CCPayment
  module Services
    class SwapService
      def initialize(client); @client = client; end
      def get_swap_coin_list; @client.post('/getSwapCoinList', {}); end
      def swap_estimate(data); @client.post('/estimate', data); end
      def get_swap_record(data); @client.post('/getSwapRecord', data); end
      def get_swap_record_list(data = {}); @client.post('/getSwapRecordList', data); end
      def swap(data); @client.post('/swap', data); end
    end
  end
end

File: php/src/Services/SwapService.php
<?php

namespace CCPayment\Services;

use CCPayment\Client;

class SwapService
{
    private Client $client;
    public function __construct(Client $client) { $this->client = $client; }
    public function getSwapCoinList(): array { return $this->client->post('/getSwapCoinList', []); }
    public function swapEstimate(array $data): array { return $this->client->post('/estimate', $data); }
    public function getSwapRecord(array $data): array { return $this->client->post('/getSwapRecord', $data); }
    public function getSwapRecordList(array $data = []): array { return $this->client->post('/getSwapRecordList', $data); }
    public function swap(array $data): array { return $this->client->post('/swap', $data); }
}
